<launch>

<arg name="map_size" value="40" />
<arg name="height_size" value="7" />
<arg name="protected_zone" value="0.29"/>
<arg name="global_resolution" value="0.20"/>
<arg name="local_map_resolution" value="$(eval 2.5 * arg('global_resolution'))"/>
<arg name="sensor_range" value="5.25"/>
<arg name="sliding_map_size" value="$(eval 4.0 * arg('sensor_range'))"/>
<arg name="mockamap_type" value="2"/>
<arg name="planning_interval" value="0.15"/>

<include file="$(find plasto)/launch/gen_mockamap.launch">
    <arg name="size" value="$(arg map_size)" />
    <arg name="height" value="$(arg height_size)" />
    <arg name="resolution" value="$(arg global_resolution)" />
    <arg name="type" value="$(arg mockamap_type)" />
</include>

<node pkg="plasto" type="plasto_node" name="plasto_node" output="screen">
    
    <remap from="/plasto_node/map" 
        to="/mock_map"/>

    <param name="ros/simulation" value="true"/>
    <param name="ros/have_global_cloud" value="true"/>
    <param name="ros/using_enu" value="false"/>
    <param name="planning/sample_tree" value="true"/>

    <param name="ros/threads" value="3"/>
    <param name="ros/simulation_step" value="30"/>
    <param name="ros/map_hz" value="15"/>

    <param name="sliding_map/size" value="$(arg sliding_map_size)"/>

    <param name="planning/runtime_error" value="0.0050"/>
    <param name="planning/refinement_time" value="0.002"/>
    <param name="planning/sensor_range" value="$(arg sensor_range)"/>
    <param name="planning/sensor_buffer_multiplier" value="2.5"/>
    <param name="planning/interval" value="$(arg planning_interval)"/>
    <param name="planning/resolution" value="$(arg protected_zone)"/>
    <rosparam param="planning/search_limit_hfov"> [0.05, 0.95] </rosparam>
    <rosparam param="planning/search_limit_vfov"> [0.10, 0.90] </rosparam>
    <param name="planning/scaled_min_dist_from_node" value="0.10"/>
    <rosparam param="planning/height"> [1.0, 3.0] </rosparam>
    <rosparam param="planning/no_fly_zone"> [] </rosparam>

    <param name="map/resolution" value="$(arg local_map_resolution)"/>
    <param name="map/size" value="$(arg map_size)"/>
    <param name="map/vfov" value="1.40"/>
    <param name="map/hfov" value="2.0944"/>
    
    <param name="amtraj/weight/time_regularization" value="1024.0"/>
    <param name="amtraj/weight/acceleration" value="15.0"/> 
    <param name="amtraj/weight/jerk" value="1.0"/>
    <!-- 0.6 -->
    <param name="amtraj/limits/max_vel" value="4.00"/>
    <param name="amtraj/limits/max_acc" value="10.00"/>
    <param name="amtraj/limits/iterations" value="23"/>
    <param name="amtraj/limits/epsilon" value="0.2"/>

    <param name="safety/total_safety_horizon" value="0.8"/>
    <param name="safety/reached_threshold" value="0.2"/>

</node>

<!-- Launch RViz with the demo configuration -->
<node name="rviz" pkg="rviz" type="rviz"
    args="-d $(find plasto)/rviz/mockamap.rviz"/>

</launch>